mmengine/registry/registry.py

```diff
@@ -142,15 +142,31 @@ def infer_scope() -> str:
             >>>     pass
             >>> # The scope of ``ResNet`` will be ``mmdet``.
         """
+        from ..logging import print_log
+
         # `sys._getframe` returns the frame object that many calls below the
         # top of the stack. The call stack for `infer_scope` can be listed as
         # follow:
         # frame-0: `infer_scope` itself
         # frame-1: `__init__` of `Registry` which calls the `infer_scope`
         # frame-2: Where the `Registry(...)` is called
-        filename = inspect.getmodule(sys._getframe(2)).__name__  # type: ignore
-        split_filename = filename.split('.')
-        return split_filename[0]
+        try:
+            filename = inspect.getmodule(  # type: ignore
+                sys._getframe(2)).__name__
+            split_filename = filename.split('.')
+            scope = split_filename[0]
+        except AttributeError:
+            # use "None" to handle some cases which can not infer the scope
+            # like initializing Registry in interactive mode
+            scope = 'None'
```

Comment:
Using `try ... except` may be ambiguous on what exactly happened. If this is intended to solve issue with interactive mode, maybe it's better to check `inspect.getmodule(...) is None` with an if-statement?

Reply:
Good suggestion. Done.